app/models/host_facets/reported_data_facet.rb

```diff
@@ -1,4 +1,25 @@
 module HostFacets
   class ReportedDataFacet < Base
+    def self.populate_fields_from_facts(host, parser, type, source_proxy)
+      facet = host.reported_data || host.build_reported_data
+      values = [
+        [:boot_time, parser.boot_timestamp],
+        [:virtual, parser.virtual],
+        [:ram, parser.ram],
+        [:sockets, parser.sockets],
+        [:cores, parser.cores],
+      ]
+      facet.attributes = Hash[values.reject { |(k, v)| v.nil? }]
```

Comment:
not crucial, just wondering - why not build a hash right away instead of an array and convert it to hash?
```suggestion
      facet.attributes = {
        boot_time: parser.boot_timestamp,
        virtual: parser.virtual,
        ram: parser.ram,
        sockets: parser.sockets,
        cores: parser.cores,
      }     
      facet.attributes.compact!
```

Reply:
because I've not realized about hash `compact` for some reason... the assignment before compact is dangerous though. Also, can we use `sym: ` instead of `=>` :rocket: ? Would be great! :heart_eyes:

Reply:
> Also, can we use `sym: ` instead of `=>` rocket ? Would be great! heart_eyes

since Ruby 1.9 IIRC :) we still use rockets in many places for consistency with existing code or when key comes from a variable, but feel free to use `sym:` where it makes sense.

Reply:
A major rewrite of all hashes creates horrible merge conflicts and makes cherry picking painful. I don't think we can enforce the new style only for new code other than suggesting it in code reviews. Perhaps we can make a post on discourse as well for better visibility.

Reply:
I didn't even know we are in favor of getting rid of :rocket:s I'm all for doing it just when we are touching the code :man_shrugging: